firstMile: Internal GO Test Findings (GO)

See feedback below from Martin Hilton

## Install Dependencies

I already have Go installed, so didn't need to follow the link. But I wonder if linking to the download page would be more helpful than the release note.

The command go get [github.com/influxdata/influxdb-client-go/v2](http://github.com/influxdata/influxdb-client-go/v2)

Resulted in this error:

go: go.mod file not found in current directory or any parent directory.
'go get' is no longer supported outside a module.
To build and install a command, use 'go install' with a version,
like 'go install [example.com/cmd@latest](http://example.com/cmd@latest)'
For more information, see https://golang.org/doc/go-get-install-deprecation
or run 'go help get' or 'go help install'.

So I improvised and created a module to continue testing with. The command worked fine there.

## Create a Token

When I generated the token (along with its warning that if I don't save it now, I'll never see it again). It covers up the instructions for what to do with the token.

## Initialize Client

No problems.

## Write Data

Attempting to run main.go I got an error:

./main.go:27:12: undefined: write

I needed to import "[github.com/influxdata/influxdb-client-go/v2/api/write](http://github.com/influxdata/influxdb-client-go/v2/api/write)" in order for it to compile.

## Execute a Simple Query

No problems

## Execute an Aggregate Query

Running main.go I got the following compile errors:

./main.go:51:8: no new variables on left side of :=
./main.go:55:15: no new variables on left side of :=

--- a/src/homepageExperience/components/HomepageExperience.scss
+++ b/src/homepageExperience/components/HomepageExperience.scss
@@ -110,5 +110,5 @@ h1#first-mile--header {
 }
 
 .homepage-wizard--code-highlight {
-  color: #B7B8FF;
+  color: #b7b8ff;
 }

--- a/src/homepageExperience/components/HomepageExperience.scss
+++ b/src/homepageExperience/components/HomepageExperience.scss
@@ -108,3 +108,7 @@ h1#first-mile--header {
   padding-right: 16px;
   max-width: 493px;
 }
+
+.homepage-wizard--code-highlight {
+  color: #B7B8FF;
+}

--- a/src/homepageExperience/components/steps/go/ExecuteAggregateQuery.tsx
+++ b/src/homepageExperience/components/steps/go/ExecuteAggregateQuery.tsx
@@ -28,7 +28,7 @@ query = \`from(bucket: "${bucket}")
               |> range(start: -10m)
               |> filter(fn: (r) => r._measurement == "measurement1")
               |> mean()\`
-results, err := queryAPI.Query(context.Background(), query)
+results, err = queryAPI.Query(context.Background(), query)
 if err != nil {
     log.Fatal(err)
 }

--- a/src/homepageExperience/components/steps/go/InstallDependencies.tsx
+++ b/src/homepageExperience/components/steps/go/InstallDependencies.tsx
@@ -43,7 +43,7 @@ export class InstallDependencies extends PureComponent {
           <SafeBlankLink href="https://go.dev/dl/">
             Go 1.17
           </SafeBlankLink>{' '}
-          or higher installed.
+          or higher installed. This sample code assumes you have go tools like <code>gofmt</code> and <code>goimports</code> installed.
         </p>
       </>
     )

--- a/src/homepageExperience/components/steps/go/InstallDependencies.tsx
+++ b/src/homepageExperience/components/steps/go/InstallDependencies.tsx
@@ -23,7 +23,8 @@ export class InstallDependencies extends PureComponent {
       <>
         <h1>Install Dependencies</h1>
         <p>
-          First, you need to create a new go module. Run the commands below in your terminal.
+          First, you need to create a new go module. Run the commands below in
+          your terminal.
         </p>
         <CodeSnippet
           text={goModuleSnippet}
@@ -31,8 +32,10 @@ export class InstallDependencies extends PureComponent {
         />
         <p style={{marginTop: '40px'}}>
           Then, install the{' '}
-          <code style={{color: '#B7B8FF'}}>influxdb-client-go</code> module. Run
-          the command below in your terminal.
+          <code className="homepage-wizard--code-highlight">
+            influxdb-client-go
+          </code>{' '}
+          module. Run the command below in your terminal.
         </p>
         <CodeSnippet
           text="go get github.com/influxdata/influxdb-client-go/v2"
@@ -40,10 +43,11 @@ export class InstallDependencies extends PureComponent {
         />
         <p style={{fontStyle: 'italic'}}>
           You’ll need to have{' '}
-          <SafeBlankLink href="https://go.dev/dl/">
-            Go 1.17
-          </SafeBlankLink>{' '}
-          or higher installed. This sample code assumes you have go tools like <code>gofmt</code> and <code>goimports</code> installed.
+          <SafeBlankLink href="https://go.dev/dl/">Go 1.17</SafeBlankLink> or
+          higher installed. This sample code assumes you have go tools like{' '}
+          <code className="homepage-wizard--code-highlight">gofmt</code> and{' '}
+          <code className="homepage-wizard--code-highlight">goimports</code>{' '}
+          installed.
         </p>
       </>
     )

--- a/src/homepageExperience/components/steps/go/InstallDependencies.tsx
+++ b/src/homepageExperience/components/steps/go/InstallDependencies.tsx
@@ -4,8 +4,18 @@ import {SafeBlankLink} from 'src/utils/SafeBlankLink'
 
 import {event} from 'src/cloud/utils/reporting'
 
+const goModuleSnippet = `mkdir -p influxdb_go_client
+cd influxdb_go_client
+go mod init influxdb_go_client
+touch main.go
+`
+
 export class InstallDependencies extends PureComponent {
-  private logCopyCodeSnippet = () => {
+  private logCopyInitializeModuleSnippet = () => {
+    event('firstMile.goWizard.initializeModule.code.copied')
+  }
+
+  private logCopyInstallCodeSnippet = () => {
     event('firstMile.goWizard.installDependencies.code.copied')
   }
   render() {
@@ -13,13 +23,20 @@ export class InstallDependencies extends PureComponent {
       <>
         <h1>Install Dependencies</h1>
         <p>
-          First, you need to install the{' '}
+          First, you need to create a new go module. Run the commands below in your terminal.
+        </p>
+        <CodeSnippet
+          text={goModuleSnippet}
+          onCopy={this.logCopyInitializeModuleSnippet}
+        />
+        <p style={{marginTop: '40px'}}>
+          Then, install the{' '}
           <code style={{color: '#B7B8FF'}}>influxdb-client-go</code> module. Run
           the command below in your terminal.
         </p>
         <CodeSnippet
           text="go get github.com/influxdata/influxdb-client-go/v2"
-          onCopy={this.logCopyCodeSnippet}
+          onCopy={this.logCopyInstallCodeSnippet}
         />
         <p style={{fontStyle: 'italic'}}>
           You’ll need to have{' '}

--- a/src/homepageExperience/components/steps/go/InstallDependencies.tsx
+++ b/src/homepageExperience/components/steps/go/InstallDependencies.tsx
@@ -23,7 +23,7 @@ export class InstallDependencies extends PureComponent {
         />
         <p style={{fontStyle: 'italic'}}>
           You’ll need to have{' '}
-          <SafeBlankLink href="https://go.dev/doc/go1.17/">
+          <SafeBlankLink href="https://go.dev/dl/">
             Go 1.17
           </SafeBlankLink>{' '}
           or higher installed.

--- a/src/homepageExperience/components/steps/go/WriteData.tsx
+++ b/src/homepageExperience/components/steps/go/WriteData.tsx
@@ -55,17 +55,17 @@ org := "${org.name}"
 bucket := "${bucket.name}"
 writeAPI := client.WriteAPIBlocking(org, bucket)
 for value := 0; value < 5; value++ {
-    tags := map[string]string{
-        "tagname1": "tagvalue1",
-    }
-    fields := map[string]interface{}{
-        "field1": value,
-    }
-    point := write.NewPoint("measurement1", tags, fields, time.Now())
+	tags := map[string]string{
+		"tagname1": "tagvalue1",
+	}
+	fields := map[string]interface{}{
+		"field1": value,
+	}
+	point := write.NewPoint("measurement1", tags, fields, time.Now())
 
-    if err := writeAPI.WritePoint(context.Background(), point); err != nil {
-        log.Fatal(err)
-    }
+	if err := writeAPI.WritePoint(context.Background(), point); err != nil {
+		log.Fatal(err)
+	}
 }`
 
   return (

--- a/src/homepageExperience/components/steps/nodejs/InstallDependencies.tsx
+++ b/src/homepageExperience/components/steps/nodejs/InstallDependencies.tsx
@@ -14,7 +14,7 @@ export class InstallDependencies extends PureComponent {
         <h1>Install Dependencies</h1>
         <p>
           First, you need to install the{' '}
-          <code style={{color: '#B7B8FF'}}>@influxdata/influxdb-client</code>{' '}
+          <code className="homepage-wizard--code-highlight">@influxdata/influxdb-client</code>{' '}
           module. Run the command below in your terminal.
         </p>
         <CodeSnippet

--- a/src/homepageExperience/components/steps/nodejs/InstallDependencies.tsx
+++ b/src/homepageExperience/components/steps/nodejs/InstallDependencies.tsx
@@ -14,7 +14,9 @@ export class InstallDependencies extends PureComponent {
         <h1>Install Dependencies</h1>
         <p>
           First, you need to install the{' '}
-          <code className="homepage-wizard--code-highlight">@influxdata/influxdb-client</code>{' '}
+          <code className="homepage-wizard--code-highlight">
+            @influxdata/influxdb-client
+          </code>{' '}
           module. Run the command below in your terminal.
         </p>
         <CodeSnippet

--- a/src/homepageExperience/components/steps/python/InstallDependencies.tsx
+++ b/src/homepageExperience/components/steps/python/InstallDependencies.tsx
@@ -14,7 +14,7 @@ export class InstallDependencies extends PureComponent {
         <h1>Install Dependencies</h1>
         <p>
           First, you need to install the{' '}
-          <code style={{color: '#B7B8FF'}}>influxdb-client</code> module. Run
+          <code className="homepage-wizard--code-highlight">influxdb-client</code> module. Run
           the command below in your terminal.
         </p>
         <CodeSnippet

--- a/src/homepageExperience/components/steps/python/InstallDependencies.tsx
+++ b/src/homepageExperience/components/steps/python/InstallDependencies.tsx
@@ -14,8 +14,10 @@ export class InstallDependencies extends PureComponent {
         <h1>Install Dependencies</h1>
         <p>
           First, you need to install the{' '}
-          <code className="homepage-wizard--code-highlight">influxdb-client</code> module. Run
-          the command below in your terminal.
+          <code className="homepage-wizard--code-highlight">
+            influxdb-client
+          </code>{' '}
+          module. Run the command below in your terminal.
         </p>
         <CodeSnippet
           text="pip3 install influxdb-client"

